### DISCUSSION TITLE
Welcome to astroid-framework Discussions!

### DISCUSSION THREAD

**Discussion by sonvnn** [templaza/astroid-framework#89] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by djayko** [comment]
Hi,

ok, let me start. I'm working with joomla almost since the beginning. I've only basic developer skills in PHP. 
I'm well experienced with HTML/CSS and SEO.
Currently I drive two public installations based on joomla 3.9 (upgrade to joomla 4 with astroid in progress) and joomla 4 (already set up with astroid).
I checked out again and again all frameworks and astroid was the best fit for me. But it can also improved more and more. ;-)

I'm really happy that astroid will still be alive. :-)

I could support the community by:
- giving ideas and suggestions
- sharing best practices
- verifying issues/new features/updates

Let's go!

best, Ayko

**Comment by djayko** [comment]
### Idea: SEO Improvements for menu banner option

## Goals:
- improve image detection for google
- reduce loading time on different devices
- usage of modern html techniques
- improve LCP (https://web.dev/lcp/)

## suggestions to discuss:
1. Use of BS5 img card option instead of background-image
Benefits:
- img srcset for different screensizes could be used -> faster loading, improved LCP
- ALT text for images could be used -> improves SEO
- aria-label for image is usable and could be connected to banner title text

Sources:
https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images
https://developers.google.com/web/fundamentals/design-and-ux/responsive/images#relative_sized_images

2.  use preload of image(s) in head
example-code:
`<link rel="preload" as="image" ....`
Benefits:
- improved loading of images

url for speed tests:
http://frank.yourweb.de/_test/astroid-test.html

Webvital-Scores:
![grafik](https://user-images.githubusercontent.com/3375044/153214654-eafd85ee-3f48-4abc-abc3-2a90b0841822.png)
![grafik](https://user-images.githubusercontent.com/3375044/153214684-a75109e9-7455-4a18-90a1-c0a38a5ae9a8.png)

Code-Pen-Playground:
https://codepen.io/djAyko/pen/XWzMZam

Just as information, why I investigated in this:
My google LCP with joomla4 on a live site (mobile view) is currently at 6.5 due to the fact that there is only one image in background used vor all screen sizes. Obviously the largest one for best image experience on desktops.


So, let's discuss what could be a useful improvement.

Best,
Ayko

**Reply by Chacapamac** [reply]
**The embedding of srcset in Astroid template manager will be so great.**

——————————————————————————
• As another idea for image, I wonder if something like Sqoosh (developp by Google) cannot be use to automate compression of images (the big speed and mobile device problem for most website) directly in Astroid 

Try it here on the web, you will be amaze of the compression — > https://squoosh.app/
Squoosh is on Github —> https://github.com/GoogleChromeLabs/squoosh
and have an API —> https://github.com/GoogleChromeLabs/squoosh/tree/dev/libsquoosh) 
and a CLI —> (https://github.com/GoogleChromeLabs/squoosh/tree/dev/cli)

**Reply by djayko** [reply]
> **The embedding of srcset in Astroid template manager will be so great.**
> 
> —————————————————————————— • As another idea for image, I wonder if something like Sqoosh (developp by Google) cannot be use to automate compression of images (the big speed and mobile device problem for most website) directly in Astroid
> 
> Try it here on the web, you will be amaze of the compression — > https://squoosh.app/ Squoosh is on Github —> https://github.com/GoogleChromeLabs/squoosh and have an API —> https://github.com/GoogleChromeLabs/squoosh/tree/dev/libsquoosh) and a CLI —> (https://github.com/GoogleChromeLabs/squoosh/tree/dev/cli)

Hi Patrick,

just for a better understanding. Your idea is using Sqoosh for image compression within astroid e.g. on image upload?
Or did you mean on upload sqoosh could be used to create different images sizes for an improved srcset implementation?

Best,
Frank

**Reply by Chacapamac** [reply]
I’m not sure if even possible
I’m just so impress with the compression scheme of Squoosh that I dream to see an integration, any integration to Joomla.

Definitively, if it can enter in the process of creating srcset images within Astroid, that will be great.
Having it as a tool to optimize images everywhere in Joomla will be fantastic, maybe as a plugin? Not sure

As we know, images are the one of the main problem on website when speed is involve. I have a life long subscription to the plugin JCH Optimize and they have a great tool to control and compress all images that can be compressed further in their plugin. I imagine that they use a internal compressor for that.

**Comment by Chacapamac** [comment]
My name is Patrick and I start to build html websites around 2002 as a hobby from my Graphic Designer and Marketing job (with 30 years + of experience). In 2008 I start to test both Wordpress & Joomla. Joomla was and still are the best existing CMS. In 2009 I embark with RocketTheme to build Joomla website. When Rocketheme change is framework to Gantry, I dislike it from the start. Gantry was pushing aside the power of Joomla core to push the Rocketheme way to do things over Joomla.

Lately, I embark in an almost 2 years deep testing of all big Joomla frameworks out there  and NONE IMPRESS ME MORE than Astroid. Astroid was lean, modern, completely Joomla friendly and with total flexibility and ease of use. It was my new champion.

Like djayko I've only basic developer skills in PHP & Javascript, I go more by intuition and well experienced with HTML/CSS and SEO.

I want to participate, because I like Joomla and the community around it. No other CMS have the same friendly and dedicated fans.
Loosing Astroid will be a blow to Joomla itself. The best CMS will loose is top best framework.

I’m going to retirement and will have more and more time to help both Joomla & Astroid:

**I can help with**
• Marketing 
• SEO
• I have a deep knowledge of Joomla & Astroid
• I’m strong in UI-UX
• bilingual french first and english (for languages files)
• I can help to test and verify issues and new features and updates

Well, I’m ready to help, and more time is coming my way to do so...

**Comment by sonvnn** [comment]
Hi @Chacapamac @Joomlaplates @djayko ,

Thank you for your support! If you love to contribute to astroidframe.work blog and tutorial. Please send me your account information via sonlv@templaza.com. I will help you create a manager account on this site to create content.

Because this site is not Allow User Registration. You need provide me (your username, email). I will help you create an account and send back to you.

Kind Regards,
Sonny

**Reply by Chacapamac** [reply]
I will apply to help for the blog, tutorial, seo, site design and content, everything, no problem.
**For sure I don’t do anything without first your approbation!**
I’m in the final for a big website right now and I will more active in the spring.

I send you my login info. Send me the password.

**_Just upload the complete french translation from the 2.5.14 english file_**

**Comment by nidzo80** [comment]
How to enable module clear all Astroid cache presented in Astroid Framework Version 2.5.13?

**Reply by joomlaplates-de** [reply]
> How to enable module clear all Astroid cache presented in Astroid Framework Version 2.5.13?

Hello,
1.) Clear Cache Module is only avaialble in Joomla 4
2.) Just download and install the latest Astroid framework > 2.5.14

**Comment by Chacapamac** [comment]
@nidzo80 
The Astroid cache button only work in Joomla 4
If you have installed 2.5.14 on Joomla 3, the cache button will not appear
You need to update to Joomla 4 and reinstall 2.5.14 over your 2.5.14 version _(just a normal drag & drop normal Joomla install)_
You Astroid/Joomla cache button should appear in top Joomla administration

**Comment by sonvnn** [comment]
We are welcome @imarklee, @mohdyasseribz and @siewho from StackIdeas come and join to this project. StackIdeas is my idol with many many amazing Joomla Extensions as EasyBlog, EasyDiscuss, EasySocial, Komento, etc. 

I am really happy when they agreed come and help us in Astroid.

Thank you very much!
Kind Regards,
Sonny

**Reply by Chacapamac** [reply]
THIS IS FANTASTIC NEWS!

**Comment by joomlaplates-de** [comment]
Anyway, 
I found out that the Astroid Backend is even more slow when you have many style copies of the theme.
Just a hint!!
Regards Peter

**Comment by nidzo80** [comment]
It's not related to Astroid but it's close. What is happening with JD Builder from JoomDev? It is still available on GitHub for J3! and works great. In my experience Astroid and JD Builder was fantastic symbiose.

**Reply by sonvnn** [reply]
Stackideas Team will be taking over the templates from Joomdev (includes JD Builder maybe).
@imarklee : May you confirm about it?

**Reply by Chacapamac** [reply]
@sonvnn  You said 
> Stackideas Team will be taking over the templates from Joomdev
**Is this mean they will participate in the new Astroid development or start their own branch?**

Just to talk, Sadly JD Builder was not “Joomla friendly”  — The best Builder was from JSN - JSN Page Builder Sadly not available in the public space.
**Anyway, I still use JCE and none of the available “Builders in the JED Today” correspond to the level of quality and are really complimenting the full power of Joomla. I test them ALL.**

JCE with it’s easy Bootstrap Columns building tools, Easy templates inserting & styles applying tools and soon to be with a block element drag and drop is 100 time leaner and more powerful than any of these builders. Also you can easily bring the the full Joomla framework and template css in the editor to make it completely WYSIWYG. _(Awesome Icons, Custom Fonts and CSS and ALL)_

**Reply by sonvnn** [reply]
> @sonvnn You said
> 
> > Stackideas Team will be taking over the templates from Joomdev
> > **Is this mean they will participate in the new Astroid development or start their own branch?**
> 
> Just to talk, Sadly JD Builder was not “Joomla friendly” — The best Builder was from JSN - JSN Page Builder Sadly not available in the public space. **Anyway, I still use JCE and none of the available “Builders in the JED Today” correspond to the level of quality and are really complimenting the full power of Joomla. I test them ALL.**
> 
> JCE with it’s easy Bootstrap Columns building tools, Easy templates inserting & styles applying tools and soon to be with a block element drag and drop is 100 time leaner and more powerful than any of these builders. Also you can easily bring the the full Joomla framework and template css in the editor to make it completely WYSIWYG. _(Awesome Icons, Custom Fonts and CSS and ALL)_

StackIdeas Team joined to this git. @imarklee is one of Admins, they will work with us in this branch.

**Comment by Chacapamac** [comment]
> will work with us in this branch
Great News!

.
